server/src/main/java/io/crate/expression/reference/doc/lucene/SourceParser.java

```diff
@@ -182,14 +171,7 @@ private static Map<String, Object> parseObject(XContentParser parser,
             XContentParser.Token token = parser.nextToken(); // move past START_OBJECT;
             for (; token == XContentParser.Token.FIELD_NAME; token = parser.nextToken()) {
                 String fieldName = lookupNameBySourceKey.apply(parser.currentName());
-                boolean dropped = false;
-                if (droppedColumns.isEmpty() == false) {
-                    String path = fieldName;
-                    if (colPath.isEmpty() == false) {
-                        path = colPath + "." + fieldName;
-                    }
-                    dropped = droppedColumns.contains(path);
-                }
+                boolean dropped = fieldName == null;
```

Comment:
Parsing became a little bit simpler, since we don't need to construct FQN anymore.